Fix statistics for binary sensor

ehendrix23 opened the pull request:
## Description:

Fix issue where statistics sensor for binary sensors does not work. Also including check to skip unknown states from statistics. Updated test_statistics to provide non-numeric values (on, off) for binary sensor test reflecting true states returned by a binary sensor.

**Related issue (if applicable):** fixes #18158

## Checklist:
  - [X] The code change is tested and works locally.
  - [X] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [X] There is no commented out code in this PR.

If the code does not interact with devices:
  - [X] Tests have been added to verify that the new code works.

ehendrix23 committed:
Fix statistics for binary sensor

-) Binary sensors have 'on' and 'off' for state resulting in issue as numbers were expected. Fixed so that it works with non-numeric states as well.
-) Added check to skip unknown states.
-) Updates test so that binary sensor test will use non-numeric values for states.

balloob commented on homeassistant/components/sensor/statistics.py:
```diff
@@ -129,11 +129,17 @@ def async_stats_sensor_startup(event):
-            self.ages.append(new_state.last_updated)
-        except ValueError:
-            pass
+        if new_state.state != STATE_UNKNOWN:
```
Can you make this a guard clause

```python

if new_state.state == STATE_UNKNOWN:
    return
```

ehendrix23 replied:
Grmbl, I should know this by now. :-)

balloob commented on homeassistant/components/sensor/statistics.py:
```diff
@@ -129,11 +129,17 @@ def async_stats_sensor_startup(event):
+
+                self.ages.append(new_state.last_updated)
+            except ValueError:
+                _LOGGER.debug("%s: adding state change '%s'", self.entity_id,
```
This should contain that it's parsing gone wrong

ehendrix23 committed:
Using guard clause and changed debug to error

Changed to use a guard clause for state unknown.
Writing error on value error instead of debug.

fabaff committed:
Add docstring

fabaff reviewed (approved):
Thanks :bird:

ehendrix23 commented:
@fabaff , thanks for fixing the issue. Was going to look at the failure this morning. :-)
Must have been a bit too liberal with my deletion when doing the previous fix :-)
